src/ai/ai.cpp

```diff
@@ -294,6 +297,12 @@ bool ai_will_accept_alliance(sys::state& state, dcon::nation_id target, dcon::na
 				}
 				++gp_count;
 			}
+			if(rel.get_are_allied()) {
+				if(alli_count >= 4) {
+					return false;
```

Comment:
This change also needs to be reflected in the offer alliance tooltip so that the player will know why they can't make an alliance

Reply:
Changed the approach to threat calculation instead